cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/detector/AnomalyDetector.java

```diff
@@ -295,6 +296,20 @@ private boolean isReadyToFix(Anomaly anomaly) {
       return false;
     }
 
+    private void mayBePersistSelfHealingInfo(Anomaly anomaly, boolean selfHealingStarted) {
+      String headerMsg = String.format("Self-healing [%s] for [%s] anomaly [%s]. Anomaly Description: [%s].",
+                                     selfHealingStarted ? "started" : "not started",
+                                     getAnomalyType(anomaly).toString(), anomaly.anomalyId(), anomaly.toString());
+      RESULT_LOG.info(headerMsg);
+      if (selfHealingStarted && anomaly instanceof KafkaAnomaly) {
```

Comment:
Why don't we use the `type` of anomaly to classify it? Today, the implementation might be based on an instance of `KafkaAnomaly`, but why do we depend on the class, but not the type?

Reply:
`getAnomalyType()` internally uses `instanceof` to get the type, it would be longer if we call `getAnomalyType(anomaly)` and then match to a type of anomaly. Since `optimizationResult()` method is defined for class `KafkaAnomaly` or `KafkaMetricAnomaly`, makes sense to check class and cast appropriately

Reply:
If `selfHealingStarted` is false here, there is no need to record optimization result. maybe the logic here can be changed to 

> 
      if (selfHealingStarted) {
        if (anomaly instanceof KafkaAnomaly) {
          RESULT_LOG.info("Optimization Result:%n{}", ((KafkaAnomaly) anomaly).optimizationResult(false));
        } else if (anomaly instanceof KafkaMetricAnomaly) {
          RESULT_LOG.info("Optimization Result:%n{}", ((KafkaMetricAnomaly) anomaly).optimizationResult(false));
        }
      }